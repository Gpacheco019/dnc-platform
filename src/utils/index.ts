export * from './localStorage';
export * from './delay';
export * from './tests/__mocks__';
export { createSetupToTests } from './tests/__config__/helpers';
