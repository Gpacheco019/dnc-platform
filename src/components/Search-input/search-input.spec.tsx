import { fireEvent } from '@testing-library/react';

import { createSetupToTests } from 'utils';

import SearchInput, { SearchInputProps } from './index';

const mockSearchInput = {
  onChange: jest.fn(),
  onSearch: jest.fn()
};

const setup = createSetupToTests<SearchInputProps>(
  { ...mockSearchInput },
  SearchInput
);

describe('<SearchInput />', () => {
  it('Should be render the component', () => {
    const { getByRole, state, getByTestId, container } = setup();
    const input = getByRole('textbox');
    const action = getByTestId('action-search-icon');

    fireEvent.change(input, { target: { value: 'teste' } });

    expect(state.onChange).toHaveBeenCalledWith('teste');

    fireEvent.click(action);
    expect(state.onSearch).toBeCalled();

    expect(container.firstChild).toMatchSnapshot();
  });
});
