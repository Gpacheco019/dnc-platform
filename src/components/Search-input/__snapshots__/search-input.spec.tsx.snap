// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> Should be render the component 1`] = `
.c0 {
  width: 24.75rem;
  height: 2.85rem;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.5rem;
}

.c1 {
  height: 100%;
  padding: 0.5rem;
  cursor: pointer;
  border-radius: 0.5rem;
}

.c1:hover {
  background-color: #EBECEE;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  padding: 0.5rem;
  border: none;
  font-weight: 300;
  border-radius: 0.5rem;
  color: #343a40;
}

<div
  class="c0"
>
  <span
    class="c1"
    data-testid="action-search-icon"
  >
    <svg
      class=""
      fill="currentColor"
      height="20"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
  </span>
  <input
    class="c2"
    maxlength="15"
    name="search"
    placeholder="Encontre seu curso..."
    type="text"
  />
</div>
`;
